specifyweb/context/app_resource.py

```diff
@@ -58,6 +58,42 @@ def get_app_resource(collection, user, resource_name, additional_default=False):
     logger.info('looking for app resource %r for user %s in %s',
                 resource_name, user and user.name, collection and collection.collectionname)
 
+    # Handling for DataObjFormatters to support fallback to defaults
+    if resource_name == 'DataObjFormatters':
+        custom_formatter = None
+        default_formatter = None
+        
+        # Try to search for the the custom formatter first
+        for level in DIR_LEVELS:
+            # First look in the database.
+            from_db = get_app_resource_from_db(collection, user, level, resource_name)
+            if from_db is not None:
+                custom_formatter = from_db
+                break
+
+            # If resource was not found, look on the filesystem.
+            from_fs = load_resource_at_level(collection, user, level, resource_name)
+            if from_fs is not None:
+                custom_formatter = from_fs
+                break
+        
+        # If custom formatter not found, try to load the default formatter from the backstop level
+        default_formatter = load_resource_at_level(collection, user, 'Backstop', resource_name)
+        
+        # If both custom and default formatters are found, merge them
+        if custom_formatter is not None and default_formatter is not None:
+            merged_formatter = merge_formatters(custom_formatter[0], default_formatter[0])
+            return merged_formatter, custom_formatter[1], custom_formatter[2]
```

Comment:
Just something to note here: 
One possible edge case here is that we merge formatters even if the `custom_formatter` was the resource found at the Backstop level (if `from_fs` was not None)

It seems `merge_formatters` does ensure we do not duplicate existing formatters so that's good! Although I do not think we would ever run into a scenario like that since most users will have a DataObjFormatter resource higher than Backstop anyway